Remapped overturning into depth space, masked the mediteranean in mom5

willaguiar opened the pull request:
Closes #274

navidcy commented:
oftentimes people say "dask gives problems" and the respond is "give us an example that does not work".

Well, here's an example that we get a lot of struggles. When I try to run this notebook to review the PR I get a bunch of 

```
2023-09-04 20:11:46,126 - distributed.utils_perf - WARNING - full garbage collections took 13% CPU time recently (threshold: 10%)
```

and more. How can we run this example smoother?

cc @aidanheerdegen, @angus-g, @dougiesquire, @micaeljtoliveira

dougiesquire commented:
One issue here is that more attention needs to be paid to the chunking of some of the data being used. E.g., adding the following to the `"mom5"` key in `psi_args` means the first few cells complete more efficiently and without warnings:

```
"chunks": {
    "time": 1,
    "potrho": -1,
    "grid_yu_ocean": 270,
    "grid_xt_ocean": -1,
 }
```

This may still not be the best chunking for the subsequent calculations, but before getting to that I'm curious why the density remapping doesn't use `xgcm`?

navidcy commented:
> This may still not be the best chunking for the subsequent calculations, but before getting to that I'm curious why the density remapping doesn't use `xgcm`?

I don't think there is a particular reason we didn't use `xgcm` but @AndyHoggANU or @adele157 may have a more definite opinion?

navidcy commented:
@willaguiar could you try @dougiesquire's suggestion here and rerun all cells?

willaguiar commented:
> One issue here is that more attention needs to be paid to the chunking of some of the data being used. E.g., adding the following to the `"mom5"` key in `psi_args` means the first few cells complete more efficiently and without warnings:
> 
> ```
> "chunks": {
>     "time": 1,
>     "potrho": -1,
>     "grid_yu_ocean": 270,
>     "grid_xt_ocean": -1,
>  }
> ```
This seems to have corrected the Dask warnings ( thanks @dougiesquire ). Could you explain in a little more detail why? ( so I keep in mind for future codes).


PS:I added a new edit to the code on the mom6 remapping function. Before, I was manually setting the southernmost cutout latitude for the remapping (78S index nmin), but the index I set before would only work on panan-01. Now I changed so the cutout (78S) latitude is automatically identified in each model.

navidcy commented:
<p>Oh I see, it's because you used <code>panant-01-zstar-ACCESSyr2</code> .</p><p>Was there any thinking behind this choice? </p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746780258'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

willaguiar commented:
<p>Added in the new commit</p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746780214'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

willaguiar commented:
<p>Thanks for noticing! Yes -  just wanted to make sure the code would work for the regional configuration. I submitted a new code where I use the global-01-v3 mom6 experiment </p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746780258'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

willaguiar commented:
<p>Same as above</p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746780295'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

willaguiar commented:
<p>done</p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746788909'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

willaguiar commented:
<p>done in the new commit</p>

---

 <a href='https://app.reviewnb.com/COSIMA/cosima-recipes/pull/284/discussion/#comment-1746788949'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

navidcy commented:
@dougiesquire I get a lot of similar-type warnings as in #285 here using unstable.

navidcy commented:
@willaguiar, can you use `pint` to deal with the unit conversion, e.g., from m^3/s -> Sv? At the moment is done ad-hoc by dividing with the conversion factor.

Have a look at https://cosima-recipes.readthedocs.io/en/latest/DocumentedExamples/Transport_Through_Straits.html

navidcy commented:
@willaguiar another remark: the ticks on the colorbar in the very last plot seems a bit weird... Am I right?

navidcy reviewed (approved):
lgtm
